src/Compilers/Core/Portable/PEWriter/ManagedResource.cs

```diff
@@ -48,7 +48,12 @@ public void WriteData(BlobBuilder resourceWriter)
                         var count = (int)(stream.Length - stream.Position);
                         resourceWriter.WriteInt32(count);
 
-                        resourceWriter.WriteBytes(stream, count);
+                        int bytesWritten = resourceWriter.TryWriteBytes(stream, count);
+                        if (bytesWritten != count)
+                        {
+                            throw new EndOfStreamException(
+                                    string.Format(CodeAnalysisResources.ResourceStreamEndedUnexpectedly, bytesWritten, count));
```

Comment:
I believe we should call an overload that takes IFormatProvider. Is the count information that valuable? Perhaps we can simply say that the stream ended unexpectedly without any numbers?